Add Route Structure, required parameters, and url matchings to backend README.md

SAbbas2018 opened the pull request:
## [DEV-006] Backend Boilerplate
<!--- Please add in the issue number this PR is fixing below. If an issue does not exist, create one! --->
Fixes: #1 

-   [Y] Has the code been formatted? (N/A)
-   [Y] Have dependencies/packages been added? (N/A)

### Changes
Added to Route structure, URL matching, and possible parameters required in the queries.

### Testing
-   [Y] Does the code have sufficient automated test coverage? If not, include a reasoning and steps required to manually QA (N/A)

SAbbas2018 committed:
Added Routes Structure to README.md

SAbbas2018 committed:
Added Delete, PUT/PATCH Routes Structure to README.md

jcserv commented on backend/README.md:
```diff
@@ -1,4 +1,60 @@
 # Pizza Backend
```
We're going by Dialog!

jcserv commented on backend/README.md:
```diff
@@ -1,4 +1,60 @@
+
+user_router := router.Group("/user"){
+
+    user_router.POST("/register", handleUserRegister) // The request respondst to the url matching "/user/register" and should include
```
move this to the next line, it's a bit difficult to read. same for the other routes!

jcserv commented on backend/README.md:
```diff
@@ -1,4 +1,60 @@
+    user_router.POST("/login", handleUserLogin) // The request responds to the url matching "/user/login" and should include
+    // parameters we require for authentication (this will be done later), we can send the user_id to the front end as a response here
+
+    Should we add a reset password for a user route when we add authentication ?
```
definitely, good thinking

jcserv commented on backend/README.md:
```diff
@@ -1,4 +1,60 @@
+
+The following are all routes for a specific user based on their user_id. Certain routes should only be accessible to professors, e.g. course creation, and archive course.
+
+specific_user_router := router.Group("/user/:user_id"){
```
can we do uid?
not as readable but it looks nicer 😎 

similarly, cid for course_id & tid for thread_id

Q: @krishchow 
is golang pothole case or camel case?
the naming for the id variables follows pothole case but in other instances of this file we use camelcase.

jcserv commented on backend/README.md:
```diff
@@ -1,4 +1,60 @@
+    // Middleware will be required to make sure only the author of the post, or an instructor is able to delete the post
+    // for author authentication we can check the user_id supplied with the author entry of the thread
+
+    specific_user_router.PUT("/:course_id/:thread_id/:comment_id/upVote", handleUpVote) // The request responds to the url matching "/users/:user_id/:course_id/:thread_id/:comment_id/upVote"
```
i think the v in upvote doesn't need to be capitallized

jcserv commented on backend/README.md:
```diff
@@ -1,4 +1,60 @@
+
+    specific_user_router.PUT("/:course_id/:thread_id/:comment_id/upVote", handleUpVote) // The request responds to the url matching "/users/:user_id/:course_id/:thread_id/:comment_id/upVote"
+    // This is an update method that allows users to upvote comments on a given post.
+    // PUT can be replaced by PATCH, since both can be used for Updating, but PATCH seems to be for modfiying, maybe use PATCH here instead?
```
some typos here:

Updating -> updating
modfiying -> modifying

in regards to the question: i agree that PATCH would be better than PUT based on the definition here:
```
PUT method uses the request URI to supply a modified version of the requested resource which replaces the original version of the resource, whereas the PATCH method supplies a set of instructions to modify the resource.
```

jcserv reviewed (changes requested):
Nice work, just a few minor fixes.
Can you also update the README.md in the root directory for the backend project structure?

SAbbas2018 committed:
Fixed issues in backend/README.md and added backend structure to root README.md

SAbbas2018 committed:
Updated terminology

SAbbas2018 committed:
Updated terminology

SAbbas2018 commented:
Fixed all the issues above. I have converted everything to **camelCase**, added the backend structure to the root README.md and added terminology in the backend/README.md.

jcserv reviewed (approved):
Great work on the fixes, I like the shorthand legend
